docs/frontend/webpack-source-maps.md

```diff
@@ -6,8 +6,402 @@ which allows a developer to view / debug developer-friendly source code instead
+[`devtool` options](https://webpack.github.io/docs/configuration.html#devtool)
+for simpler / development configuration. In our case here, we:
+
+- Enable minification with the `UglifyJsPlugin` so that we have a good example
+  to see the impact of source maps with.
+- Override the `append` option to `SourceMapDevToolPlugin` so that we can insert
+  a full URL in the outputted `//# sourceMappingURL` comment.
+
+[`webpack.config.js`](../../examples/frontend/webpack-source-maps/webpack.config.js)
+
+```js
+var path = require("path");
+var webpack = require("webpack");
+
+module.exports = {
+  context: path.join(__dirname, "../src/es5"),
+  entry: {
+    app1: "./app1.js",
+    app2: "./app2.js"
+  },
+  output: {
+    path: path.join(__dirname, "dist/js"),
+    filename: "[name].js"
+  },
+  plugins: [
+    new webpack.optimize.UglifyJsPlugin({
+      // Preserve the source map comment in minified code.
+      sourceMap: true
+    }),
+    new webpack.SourceMapDevToolPlugin({
+      // https://webpack.github.io/docs/configuration.html#output-sourcemapfilename
+      filename: "[name].js.map",
+
+      // Output sourceMappingURL comment.
+      //
+      // Examples:
+      // 1. Default to: `"\n//# sourceMappingURL=[url]"` for relative hosting
+      append: ""
+      // 2. Localhost + local repo checkout.
+      // append: "\n//# sourceMappingURL=http://localhost:3000/" +
+      //   "examples/frontend/webpack-source-maps/dist/js/[url]"
+      // 3. Internal VPN URL.
+      // append: "\n//# sourceMappingURL=http://my-vpn-url.com/PATH/[url]"
+    })
+  ]
+};
+```
+
+This produces four files:
+
+* [`dist/js/app1.js`](../../examples/frontend/webpack-source-maps/dist/js/app1.js):
+  The `app1` entry point.
+* [`dist/js/app1.js.map`](../../examples/frontend/webpack-source-maps/dist/js/app1.js.map):
+  The `app1` source map file.
+* [`dist/js/app2.js`](../../examples/frontend/webpack-source-maps/dist/js/app1.js):
+  The `app2` entry point.
+* [`dist/js/app2.js.map`](../../examples/frontend/webpack-source-maps/dist/js/app2.js.map):
+  The `app2` source map file.
+
+Let's look at the `app1` files in detail (the `app2` files are analogous):
+
+[`dist/js/app1.js`](../../examples/frontend/webpack-source-maps/dist/js/app1.js)
+
+```js
+!function(n){function r(e){/*... SNIPPED ...*/document.querySelector("#content").innerHTML+=e("app1","App 1")}]);
+//# sourceMappingURL=app1.js.map
+```
+
+The `app1.js` file contains a minified bundle of the `app1` entry point with the
+key `sourceMappingURL` control comment at the bottom. As it stands now, an
+`app1.js.map` file is expected to be served from the same directory as the
+application bundle.
+
+As discussed in the [Integrating Source Maps](#integrating-source-maps) section,
+there are many ways to serve and write the `sourceMappingURL`. Here, we take
+the easy way out and choose "Option 1" of serving the maps alongside code.
+
+While this is fine for this playbook example, real world usage would tend
```

Comment:
I'm a broken record at this point, but if I don't do it the "real world" way, what am I sacrificing? Just IP protection? Calling one "real world" seems to imply that the other is just for toys.